modules/launcher/src/main/java/org/mule/module/launcher/descriptor/ArtifactDescriptor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.launcher.descriptor;
+
+import org.mule.module.launcher.artifact.Artifact;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Represents an abstract description of an {@link Artifact}
```

Comment:
Just that's the basis of the rest. Probably "base description" is more accurate.